### ISSUE TITLE
rust-analyzer.rustfmt.overrideCommand doesn't seem to work

### ISSUE THREAD

**Issue by jplatte** [rust-lang/rust-analyzer#13172]
The current documentation on this setting is not very clear:

> Advanced option, fully override the command rust-analyzer uses for formatting.

From #10826 I gathered that I should be able to set it to an array containing an absolute path to a rustfmt executable, but this just caused rust-analyzer to not format files at all anymore (I set it to my nightly toolchain's rustfmt binary).

Could it be documented how this option is meant to be used? Alternatively, it would be great if I could just tell rust-analyzer to use rustfmt from a different toolchain than what's used for everything else.

**Comment by jplatte** [comment]
Well, now I tried again and it just worked (regardless of whether the edition was passed or not). Can I leave this open as a documentation improvement request?

**Comment by jplatte** [comment]
Hm no, it's back to not functioning. No logs to be found other than some initialization logs in the Rust Analyzer Client section.

**Comment by Veykril** [comment]
Do you have the `RA_LOG` env var overwritten? If not this should at least log an error ... does running rustfmt manually work on the file you are trying to format?

**Comment by jplatte** [comment]
No, I haven't set `RA_LOG`. I'll try with manual `rustfmt` invocation in a bit.

**Comment by Veykril** [comment]
Iirc there is still a bug where very weird whitespace can make rustfmt spit out a warning, exiting successfully without formatting anything. This might be what you are running into, at least I hope that's it 😬.

**Comment by jplatte** [comment]
Seems to work more reliable now I've enabled it again. So only the documentation is left.

**Comment by andrewbaxter** [comment]
Just landed here.  Yeah, the documentation isn't very informative, I assume since this is only intended to work with rustfmt also.

In short

- args are an array, or null/absent to use default rustfmt
- first element is binary
- other arguments are arguments to the binary
- the source code is passed in via stdin
- the formatted code is taken from stdout

If this were in the docs it would be awesome.
